src/platform/packages/shared/kbn-esql-validation-autocomplete/src/shared/resources_helpers.ts

```diff
@@ -19,13 +19,18 @@ import type { ESQLCallbacks } from './types';
 
 export const NOT_SUGGESTED_TYPES = ['unsupported'];
 
+// Helper to check if a command is a source command that should be cached
+function isSourceCommand(commandName: string): boolean {
+  return SOURCE_COMMANDS.has(commandName.toLowerCase());
+}
+
 /**
  * Responsible for efficiently computing the list of columns generated by a specific ESQL query.
  */
 export class QueryColumns {
   private static readonly cache = new Map<string, ESQLColumnData[]>();
   // Adding a max size to the cache to prevent unbounded memory growth
-  private static readonly MAX_CACHE_SIZE = 200;
+  private static readonly MAX_CACHE_SIZE = 100;
```

Comment:
Back to 100 as now we are only caching source commands to solve the multiple network requests problem